internal/hcsoci/create.go

```diff
@@ -229,3 +247,108 @@ func CreateContainer(ctx context.Context, createOptions *CreateOptions) (_ cow.C
 	}
 	return system, resources, nil
 }
+
+// CloneContainer is similar to CreateContainer but it does not add layers or namespace like
+// CreateContainer does. Also, instead of sending create container request it sends a modify
+// request to an existing container. CloneContainer only works for WCOW.
+func CloneContainer(ctx context.Context, createOptions *CreateOptions) (_ cow.Container, _ *Resources, err error) {
+	coi, err := initializeCreateOptions(ctx, createOptions)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	if coi.Spec.Windows == nil || coi.HostingSystem == nil {
+		return nil, nil, fmt.Errorf("CloneContainer is only supported for Hyper-v isolated WCOW ")
+	}
+
+	resources := &Resources{
+		id: createOptions.ID,
+	}
+	defer func() {
+		if err != nil {
+			if !coi.DoNotReleaseResourcesOnFailure {
+				ReleaseResources(ctx, resources, coi.HostingSystem, true)
+			}
+		}
+	}()
+
+	if coi.HostingSystem != nil {
+		n := coi.HostingSystem.ContainerCounter()
+		if coi.Spec.Linux != nil {
+			resources.containerRootInUVM = fmt.Sprintf(lcowRootInUVM, createOptions.ID)
+		} else {
+			resources.containerRootInUVM = fmt.Sprintf(wcowRootInUVM, strconv.FormatUint(n, 16))
+		}
+	}
+
+	if err = setupMounts(ctx, coi, resources); err != nil {
+		return nil, resources, err
+	}
+
+	// TODO(ambarve): everything that is added to the container during the createContainer
+	// request (via the gcsDocument) must be hot added here.
+	// 1. All types of mounts - scsi, physical disk, vsmb, pipes
+	// 2. Networking
+	// 3. GMSA credentials?
```

Comment:
GMSA only for process isolated for the time being so I wouldn't worry about this for now.